packages/classic-webview/src/components/UnlockHardcoreModal.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import { HardcoreLogo } from '@hotandcold/webview-common/components/logo';
+import { GoldIcon, CloseIcon } from '@hotandcold/webview-common/components/icon';
+import { cn } from '@hotandcold/webview-common/utils';
+import { useModal } from '../hooks/useModal';
+
+interface PurchaseButtonProps {
+  children: React.ReactNode;
+  style: 'primary' | 'secondary';
+  price: number;
+  onClick?: () => void;
+}
+
+const PurchaseButton: React.FC<PurchaseButtonProps> = (props) => {
+  const { children, price, onClick, style } = props;
+
+  return (
+    <button
+      onClick={onClick}
+      className={cn(
+        'flex w-full flex-row items-center justify-between gap-4 whitespace-nowrap rounded-full border-2 border-current px-6 py-3 text-center font-sans sm:w-auto',
+        style === 'primary'
+          ? 'border-mustard-gold text-mustard-gold'
+          : 'border-slate-gray text-slate-gray'
+      )}
+    >
+      <span className="text-left text-base font-semibold">{children}</span>
+      <span
+        className={cn(
+          'flex w-fit flex-row items-center gap-[6px] rounded-full px-3 py-2 text-xs font-semibold',
+          style === 'primary' ? 'bg-mustard-gold text-black' : 'bg-charcoal text-white'
+        )}
+      >
+        <span className="flex h-4 w-4 items-center justify-center">
+          <GoldIcon />
+        </span>
+        <span className="flex items-center">Use {price}</span>
```

Comment:
This span seems unnecessary
```suggestion
       Use {price}
```